### ISSUE TITLE
Known issue: Presence calculation doesn't currently handle multiple departures within one day where the last event is a departure

### ISSUE THREAD

**Issue by robholmesnz** [ServiceInnovationLab/PresenceChecker#175]
Resolution is likely to involve creating tests which break current calculator and making changes so that these pass.

**Comment by Br3nda** [comment]
Probably a days' work would get the calculation right

**Comment by CharlotteHinton** [comment]
Descoped from a discussion with Rob.

**Comment by mischa-s** [comment]
@Br3nda this work would need to happen in Open Fisca right?

**Comment by Br3nda** [comment]
No, it is in presence checker. PC works out the `present_in_nz` variable from movement data and sends that as input to openfisca. 

Presence checker needs to correctly work out this variable when there are many movements in a day.

Consider the following:

Arriving, then leaving on the same day means you are present today (for one day), and not present tomorrow, and you stayed absent from then on. 

Arriving, leaving, then arriving again all on the same day means you are present today and tomorrow, and stayed present after. 

Leaving then arriving back today (or tomorrow) means you are present both days, and stayed present. 

Leaving, arriving, then leaving on the same day means you are present today, but absent tomorrow and stay absent from then on


Alternative description:
 work out their last movement on a day, and apply the appropriate state to tomorrow 
(unless there are more movements to consider tomorrow)

**Comment by mischa-s** [comment]
okay, I might do a little spike on this then as it does feel quite important

**Comment by mischa-s** [comment]
probably looks like a refactor of /app/services/eligibility_service.rb  97 - 102
